### ISSUE TITLE
can't get it to do... anything?

### ISSUE THREAD

**Issue by wolffcatskyy** [projectstorm/dynamouse#4]
I have a four monitor setup on a Mac Pro 6,1 macOS 14.3.1 (Open Core Legacy Patcher).  I installed your 1.0 release, gave it permissions for accessibility, input monitoring, and automation.  The app ran fine, identified my devices and screens, and allowed me to assign devices to screens.  However when I moved my mice they still moved across all screens.  I  rebooted, reinstalled, even uninstalled and downgraded to your first release, but nothing worked.

In summary, it seems to install and operate, but it doesn't DO anything.

Logitech USB keyboard mouse
Apple Bluetooth Mouse

PS. Really ant this to work thanks for your efforts!

**Comment by dylanvorster** [comment]
Ive also got a system with Open Core Legacy Patcher (and am running into some issues as well in general). Im busy adding some better logging now, and going to test some more on my side. Out of interest, it may be worth try the previous version while I try and diagnose whats going on.

**Comment by wolffcatskyy** [comment]
Thanks for the quick response. I thought I wrote that I did in fact try the
previous version as well. Same thing nothing.

Please let me know if I can do anything to help your testing or if you’d
like to do a remote desktop or anything else

**Comment by dylanvorster** [comment]
Sorry my mistake, my brain went into auto-try-suggest-solutions-mode without even thinking. I have a new version which I think might fix the issue. I discovered that multiple HID devices were being detected for the same device instance but with different usagePage identifiers. The final group-by logic could result in the wrong one being picked. I tested the new version (1.1.0) and my own issues I have been running into seem to be solved, even when I reboot with zero startup-time delay configured. If you try the newer version shortly, I would disable and then re-enable the automatic startup mode which will add the 1.1.0 version to the startup items. Ive also added file logging which can be enabled that should log to the ~/Library/logs/@projectstorm/dynamouse directory (in debug mode). Feel free to share that file if you are are still running into issues with this new version (and are ofc happy sharing it here).

If you still run into issues, ill fire up my old 2014 macbook pro running Open Core and then will take a look.

Thanks again for helping me make this utility better by sharing your usage of it :)

**Comment by dylanvorster** [comment]
(busy with the release now)

**Comment by wolffcatskyy** [comment]
Wow that was fast!

On my Mac, I have a Logitech Bolt USB Receiver and a USB trackball.  I thought it would detect both, but it seems to be only one or the other.

When I assigned a mouse to ba screen, it moved the cursor there, but it wasn’t locked to the screen as expected.

Here’s the log.  Thanks!
￼

**Comment by wolffcatskyy** [comment]
The functionality I am looking for is essentially the same as MouseMux:

https://www.mousemux.com/￼

I know your app isn’t identical but shares some aspects.  The use case is two simultaneous users with their own mouse, keyboard, and cursor for collaboration.  There are a lot of tools to do this remotely, but I can’t find anything on a Mac that will allow it locally.  With the power of the current generation of Macs, they should easily be able to handle a couple simultaneous web browsing sessions right?

**Comment by dylanvorster** [comment]
Hmm okay so theres a few things here I think:

` I thought it would detect both, but it seems to be only one or the other` it sounds like its working now (at least within the current feature set of DynaMouse) (this is how I use it in my music studio).

`but it wasn’t locked to the screen as expected` So this is also by design, since I need to be able to drag windows across to different screens. I do however think this could be a useful addition to the software via a configuration setting on each mapping (I will put that below the 'None' option).

`The functionality I am looking for is essentially the same as MouseMux` So this is a much harder problem in many aspects because it involves quite a few different things:
* multiple cursors on OSX - this is not possible natively to the best of my knowledge, but could be implemented via a glass-pane-like transparent window that renders a transparent canvas on each screen and then draws a virtual mouse.
* `but I can’t find anything on a Mac that will allow it locally` interesting use case, Im not sure though if I would have the time to implement this using the method I described above. If there are more efficient ways of drawing smaller widgets on OSX, it would probably need to be done with a more native language such as swift or Objective-C. Swing, AWT and many other widget frameworks I'm familiar with tend to lock their drawing capability inside window and im not sure how seamless I could make it. I also wrote this in Typescript after getting frustrated with Swift meaning the implementation I do in Dynamouse would be web based, which may not have the fastest and best performance.
